### PR TITLE
Use command base name not full path.

### PR DESCRIPTION
**Pull request by mboetger**
Fix for https://github.com/google/argh/issues/12

### CODE REVIEW

**Comment by mboetger** [comment]
@googlebot I signed it!

File: src/lib.rs [hunk 1]
```diff
@@ -255,6 +260,11 @@ pub fn from_env<T: TopLevelCommand>() -> T {
 /// was unsuccessful or if information like `--help` was requested.
 pub fn cargo_from_env<T: TopLevelCommand>() -> T {
```
Comment (benbrittain) [inline src/lib.rs:261]: we should probably remove this function honestly

Comment (benbrittain) [reply src/lib.rs:261]: not expecting than in this CL, just making a note for me

File: src/lib.rs [hunk 2]
```diff
@@ -236,6 +236,11 @@ impl From<String> for EarlyExit {
 /// was unsuccessful or if information like `--help` was requested.
 pub fn from_env<T: TopLevelCommand>() -> T {
     let strings: Vec<String> = std::env::args().collect();
+    let cmd = std::path::Path::new(&strings[0])
```
Comment (benbrittain) [inline src/lib.rs:239]: this doesn't seem to be used?

Comment (mboetger) [reply src/lib.rs:239]: Oops. I made a new branch with the changes.  Missed that.

Comment (mboetger) [reply src/lib.rs:239]: Ok, being used now.

**Comment by benbrittain** [comment]
ok! I'm just used to that poping up when people accidentally sign it :)

**Comment by benbrittain** [comment]
I'd like a test before merging, but other than that lgtm

**Comment by erickt** [comment]
@mboetger would you be able to add some tests for this?

**Review by mboetger — COMMENTED** [review]
Added unit tests.  Sorry for the noisey merge.

File: src/lib.rs [hunk 3]
```diff
@@ -230,14 +230,20 @@ impl From<String> for EarlyExit {
     }
 }
 
+/// Extract the base cmd from a path
+pub fn cmd<'a>(default: &'a String, path: &'a String) -> &'a str {
```
Comment (erickt) [inline src/lib.rs:234]: I don't think this function should be part of the public. Could you make this private, and move the test you defined in `tests/lib.rs` into `src/lib.rs`, and add it like:

```
...
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_cmd_extraction() {
        let expected = "test_cmd";
        let path = format!("/tmp/{}", expected);
        let cmd = argh::cmd(&path, &path);
        assert_eq!(expected, cmd);
    }
}
```

**Review by erickt — CHANGES_REQUESTED** [review]
other than that, LGTM

**Review by erickt — APPROVED** [review]

**Pull request merged**